packages/experiments/src/components/Stack/Stack.styles.ts

```diff
@@ -14,24 +15,49 @@ export const styles = (props: IThemedProps<IStackProps>): IStackStyles => {
     maxHeight,
     horizontal,
     grow,
-    margin,
+    wrap,
     padding,
     horizontalAlignment,
-    verticalAlignment
+    verticalAlignment,
+    horizontalGap,
+    verticalGap,
+    shrinkItems,
+    theme,
+    className
   } = props;
 
+  const hGap = parseGap(horizontalGap, theme);
+  const vGap = parseGap(verticalGap, theme);
+
+  // styles to be applied to all direct children regardless of wrap or direction
+  const childStyles = {
+    whiteSpace: 'nowrap',
+    textOverflow: 'ellipsis'
+  };
+
+  // selectors to be applied regardless of wrap or direction
+  const commonSelectors = {
+    '> *:empty': {
+      display: 'none'
+    },
+
+    // flexShrink styles are applied by the StackItem
+    '> *:not(.ms-StackItem)': {
+      flexShrink: shrinkItems ? 1 : 0
+    }
+  };
+
   return {
     root: [
       {
         display: 'flex',
         flexDirection: horizontal ? 'row' : 'column',
         flexWrap: 'nowrap',
-        width: fillHorizontal ? '100%' : 'auto',
-        height: fillVertical ? '100%' : 'auto',
+        width: fillHorizontal && !wrap ? '100%' : 'auto',
```

Comment:
total nitpick but i find adding parenthesis here helps to resolve expectations:

Could be one of these:

```tsx
(fillHorizontal && !wrap) ? '100%' : 'auto'
fillHorizontal && (!wrap ? '100%' : 'auto')
```